pytential/symbolic/stokes.py

```diff
@@ -469,36 +253,443 @@ def apply_stress(self, density_vec_sym, normal_vec_sym, dir_vec_sym,
+                        qbx_forced_limit, deriv_dirs=extra_deriv_dirs)
+
+        return np.array(rewrite_using_base_kernel(sym_expr,
+            base_kernel=self.base_kernel))
+
+    def apply_stokeslet_and_stresslet(self, stokeslet_density_vec_sym,
+            stresslet_density_vec_sym, dir_vec_sym,
+            qbx_forced_limit, stokeslet_weight, stresslet_weight,
+            extra_deriv_dirs=()):
+
+        stokeslet_obj = StokesletWrapper(dim=self.dim,
+                mu_sym=self.mu, nu_sym=self.nu, method=self.method)
+
+        sym_expr = 0
+        if stresslet_weight != 0:
+            sym_expr += self.apply(stresslet_density_vec_sym, dir_vec_sym,
+                qbx_forced_limit, extra_deriv_dirs) * stresslet_weight
+        if stokeslet_weight != 0:
+            sym_expr += stokeslet_obj.apply(stokeslet_density_vec_sym,
+                qbx_forced_limit, extra_deriv_dirs) * stokeslet_weight
+
+        return sym_expr
+
+    def apply_stress(self, density_vec_sym, normal_vec_sym, dir_vec_sym,
+                        qbx_forced_limit):
 
         sym_expr = np.empty((self.dim,), dtype=object)
 
         # Build velocity derivative matrix
         sym_grad_matrix = np.empty((self.dim, self.dim), dtype=object)
         for i in range(self.dim):
             sym_grad_matrix[:, i] = self.apply_derivative(i, density_vec_sym,
-                                     normal_vec_sym, mu_sym, qbx_forced_limit)
+                                     normal_vec_sym, qbx_forced_limit)
 
         for comp in range(self.dim):
 
             # First, add the pressure term:
             sym_expr[comp] = - dir_vec_sym[comp] * self.apply_pressure(
                                             density_vec_sym, normal_vec_sym,
-                                            mu_sym, qbx_forced_limit)
+                                            qbx_forced_limit)
 
             # Now add the velocity derivative components
             for j in range(self.dim):
                 sym_expr[comp] = sym_expr[comp] + (
-                                    dir_vec_sym[j] * mu_sym * (
+                                    dir_vec_sym[j] * self.mu * (
                                         sym_grad_matrix[comp][j]
                                         + sym_grad_matrix[j][comp])
                                         )
 
         return sym_expr
 
+
+class StressletWrapperNaive(_StressletWrapperNaiveOrBiharmonic):
+    def __init__(self, dim=None, mu_sym=_MU_SYM_DEFAULT, nu_sym=0.5):
+        super().__init__(dim=dim, mu_sym=mu_sym, nu_sym=nu_sym,
+                method="naive")
+
+
+class StressletWrapperBiharmonic(_StressletWrapperNaiveOrBiharmonic):
+    def __init__(self, dim=None, mu_sym=_MU_SYM_DEFAULT, nu_sym=0.5):
+        super().__init__(dim=dim, mu_sym=mu_sym, nu_sym=nu_sym,
+                method="biharmonic")
+
+# }}}
+
+
+# {{{ Stokeslet/Stresslet using Laplace (Tornberg)
+
+class StokesletWrapperTornberg(StokesletWrapperBase):
```

Comment:
This doesn't implement `apply_stress` :cry:

Reply:
The paper doesn't mention how to do it. Let's open an issue.

Reply:
Shouldn't this work like for the biharmonic case? i.e. just use the `StressletWrapperTornberg` and do `d_i S_{ijk}`.
https://github.com/isuruf/pytential/blob/9e07ac228a3a5289249ff4d3015a0c9b506a2f2f/pytential/symbolic/stokes.py#L347-L363

I'm not very familiar with that paper, so maybe there's a more efficient way to construct it? Either way, leaving it for an issue / follow-up PR is fine!